lib/http-bridge/test/integration/Test/Integration/Scenario/Transactions.hs

```diff
@@ -309,3 +353,372 @@ spec = do
+                }|]
+            let endpoint = "v2/wallets/" <> w ^. walletId <> "/transactions"
+            r <- request @(ApiTransaction HttpBridge) ctx (method, endpoint)
+                    Default payload
+            expectResponseCode @IO HTTP.status405 r
+            expectErrorMessage errMsg405 r
+
+    describe "TRANS_CREATE_08 - HTTP headers" $ do
+        let matrix =
+                 [ ( "No HTTP headers -> 415", None
+                   , [ expectResponseCode @IO HTTP.status415
+                     , expectErrorMessage errMsg415 ]
+                   )
+                 , ( "Accept: text/plain -> 406"
+                   , Headers
+                         [ ("Content-Type", "application/json")
+                         , ("Accept", "text/plain") ]
+                   , [ expectResponseCode @IO HTTP.status406
+                     , expectErrorMessage errMsg406 ]
+                   )
+                 , ( "No Accept -> 202"
+                   , Headers [ ("Content-Type", "application/json") ]
+                   , [ expectResponseCode @IO HTTP.status202 ]
+                   )
+                 , ( "No Content-Type -> 415"
+                   , Headers [ ("Accept", "application/json") ]
+                   , [ expectResponseCode @IO HTTP.status415
+                     , expectErrorMessage errMsg415 ]
+                   )
+                 , ( "Content-Type: text/plain -> 415"
+                   , Headers [ ("Content-Type", "text/plain") ]
+                   , [ expectResponseCode @IO HTTP.status415
+                     , expectErrorMessage errMsg415 ]
+                   )
+                 ]
+        forM_ matrix $ \(title, headers, expectations) -> it title $ \ctx -> do
+            w <- fixtureWallet ctx
+            wDest <- emptyWallet ctx
+            addr:_ <- listAddresses ctx wDest
+            let destination = addr ^. #id
+            let payload = Json [json|{
+                    "payments": [{
+                        "address": #{destination},
+                        "amount": {
+                            "quantity": 1,
+                            "unit": "lovelace"
+                        }
+                    }],
+                    "passphrase": "cardano-wallet"
+                }|]
+            r <- request @(ApiTransaction HttpBridge) ctx (postTx w)
+                    headers payload
+            verify r expectations
+
+    describe "TRANS_CREATE_08 - Bad payload" $ do
+        let matrix =
+                [ ( "empty payload", NonJson "" )
+                , ( "{} payload", NonJson "{}" )
+                , ( "non-json valid payload"
+                  , NonJson
+                        "{ payments: [{\
+                         \\"address\": 12312323,\
+                         \\"amount: {\
+                         \\"quantity\": 1,\
+                         \\"unit\": \"lovelace\"} }],\
+                         \\"passphrase\": \"cardano-wallet\" }"
```

Comment:
It took me a good 15 seconds to figure out what was wrong with this payload :sweat_smile:  ... I'd suggest to make it a bit shorter, is doesn't need much more than `"{ payments: []}"` in the end to make it non-valid JSON.

Reply:
Yes, however `{ payments: []}` is not as fun as this one... ;-)
I mean it's pretty obvious to fail and most probably always would... this one is not so obvious, so _if_ it would fail, that'd be fun... :-)